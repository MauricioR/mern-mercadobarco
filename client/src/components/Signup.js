import React, { Component } from 'react';
import { Button, Form, FormGroup, Label, Input, FormText } from 'reactstrap';

class Signup extends Component {
  render() {
    return (
      <Form>
        <FormGroup>
          <Label for="firstName">First Name</Label>
          <Input type="text" name="name" id="firstName" />
        </FormGroup>
        <FormGroup>
          <Label for="lastName">Last Name</Label>
          <Input type="text" name="name" id="lastName" />
        </FormGroup>
        <FormGroup>
          <Label for="email">Email</Label>
          <Input type="email" name="email" id="email" />
        </FormGroup>
        <FormGroup>
          <Label for="password">Password</Label>
          <Input type="password" name="password" id="password" />
        </FormGroup>
        <Button>Submit</Button>
      </Form>
    );
  }
}

export default Signup;
