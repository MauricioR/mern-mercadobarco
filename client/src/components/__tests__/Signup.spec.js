import React from 'react';
import { shallow, mount } from 'enzyme';

import { Button, Form, FormGroup, Label, Input, FormText } from 'reactstrap';

import Signup from '../Signup';

describe('Testing Signup Component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Signup />);
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render firstName label', () => {
    expect(wrapper.find(Label).filterWhere((item) => {
      return item.prop('for') === 'firstName';
    })).toHaveLength(1);
  });

  it('should render firstName input', () => {
    expect(wrapper.find(Input).filterWhere((item) => {
      return item.prop('id') == 'firstName';
    })).toHaveLength(1);
  });

  it('should render lastName label', () => {
    expect(wrapper.find(Label).filterWhere((item) => {
      return item.prop('for') === 'lastName';
    })).toHaveLength(1);
  });

  it('should render lastName input', () => {
    expect(wrapper.find(Input).filterWhere((item) => {
      return item.prop('id') == 'lastName';
    })).toHaveLength(1);
  });

  it('should render email label', () => {
    expect(wrapper.find(Label).filterWhere((item) => {
      return item.prop('for') === 'email';
    })).toHaveLength(1);
  });

  it('should render email input', () => {
    expect(wrapper.find(Input).filterWhere((item) => {
      return item.prop('id') == 'email';
    })).toHaveLength(1);
  });

  it('should render password label', () => {
    expect(wrapper.find(Label).filterWhere((item) => {
      return item.prop('for') === 'password';
    })).toHaveLength(1);
  });

  it('should render password input', () => {
    expect(wrapper.find(Input).filterWhere((item) => {
      return item.prop('id') == 'password';
    })).toHaveLength(1);
  });

  it('should render submit button', () => {
    expect(wrapper.find(Button).at(0).text()).toEqual("Submit");
  });
});
