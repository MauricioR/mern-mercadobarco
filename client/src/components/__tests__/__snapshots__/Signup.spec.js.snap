// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Signup Component should render correctly 1`] = `
<Signup>
  <Form
    tag="form"
  >
    <form
      className=""
    >
      <FormGroup
        tag="div"
      >
        <div
          className="position-relative form-group"
        >
          <Label
            for="firstName"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <label
              className=""
              htmlFor="firstName"
            >
              First Name
            </label>
          </Label>
          <Input
            id="firstName"
            name="name"
            type="text"
          >
            <input
              className="form-control"
              id="firstName"
              name="name"
              type="text"
            />
          </Input>
        </div>
      </FormGroup>
      <FormGroup
        tag="div"
      >
        <div
          className="position-relative form-group"
        >
          <Label
            for="lastName"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <label
              className=""
              htmlFor="lastName"
            >
              Last Name
            </label>
          </Label>
          <Input
            id="lastName"
            name="name"
            type="text"
          >
            <input
              className="form-control"
              id="lastName"
              name="name"
              type="text"
            />
          </Input>
        </div>
      </FormGroup>
      <FormGroup
        tag="div"
      >
        <div
          className="position-relative form-group"
        >
          <Label
            for="email"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <label
              className=""
              htmlFor="email"
            >
              Email
            </label>
          </Label>
          <Input
            id="email"
            name="email"
            type="email"
          >
            <input
              className="form-control"
              id="email"
              name="email"
              type="email"
            />
          </Input>
        </div>
      </FormGroup>
      <FormGroup
        tag="div"
      >
        <div
          className="position-relative form-group"
        >
          <Label
            for="password"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <label
              className=""
              htmlFor="password"
            >
              Password
            </label>
          </Label>
          <Input
            id="password"
            name="password"
            type="password"
          >
            <input
              className="form-control"
              id="password"
              name="password"
              type="password"
            />
          </Input>
        </div>
      </FormGroup>
      <Button
        color="secondary"
        tag="button"
      >
        <button
          className="btn btn-secondary"
          onClick={[Function]}
        >
          Submit
        </button>
      </Button>
    </form>
  </Form>
</Signup>
`;
